tensorboard/webapp/metrics/views/main_view/common_selectors_test.ts

```diff
@@ -181,118 +169,102 @@ describe('common selectors', () => {
         },
       } as any,
       hparams: {
-        specs: buildSpecs('defaultExperimentId', {
-          hparam: {
-            specs: [buildHparamSpec({name: 'foo', displayName: 'Foo'})],
-            defaultFilters: new Map(),
-          },
-          metric: {
-            specs: [buildMetricSpec({displayName: 'Bar'})],
-            defaultFilters: new Map(),
-          },
-        }),
+        dashboardSpecs: {
+          hparams: [buildHparamSpec({name: 'foo', displayName: 'Foo'})],
+          metrics: [buildMetricSpec({displayName: 'Bar'})],
+        },
+        dashboardSessionGroups: [],
       } as any,
     });
   });
 
   describe('getScalarTagsForRunSelection', () => {
     it('returns all tags containing scalar data when no runs are selected', () => {
-      const state = {
-        ...appStateFromMetricsState(
-          buildMetricsState({
-            tagMetadata: {
-              histograms: {
-                tagDescriptions: {},
-                tagToRuns: {},
-              },
-              images: {
-                tagDescriptions: {},
-                tagRunSampledInfo: {},
-              },
-              scalars: {
-                tagDescriptions: {},
-                tagToRuns: {
-                  'tag-1': ['run1'],
-                  'tag-2': ['run2', 'run3'],
-                },
-              },
+      const state = buildMockState({
+        metrics: buildMetricsState({
```

Comment:
buildMockState is useful but it doesn't change the fact that we can continue to use appStateFromMetricsState, buildStateFromAppRoutingState, etc..

Instead you have changed to referring to METRICS_FEATURE_KEY, APP_ROUTING_FEATURE_KEY, etc by their bare constant values 'metrics', 'app_routing', etc... but that seems unnecessary and mildly controversial?

There could be some pros to do this (readability, for instance) but there are also clearly some cons (maintainability). It would be worth consulting the rest of the team about this. Maybe split those particular changes from this PR and we can discuss at the next FE team?